File: lecture notes/matlab/lec6/run1.m
clear;clc;

figure(1);clf;hold on;grid minor;set(gca, 'MinorGridColor', 'k','MinorGridAlpha',1);
xlabel("Re");ylabel("Im");title("w_n sabit ve 0<\zeta<1");
T=1;
zetavec=0.1:0.1:1;
wnvec=1:6;
for i=1:length(wnvec)
    wn=wnvec(i);
    fx=exp(-zetavec*wn*T).*cos(sqrt(1-zetavec.^2)*wn*T);
    fy=exp(-zetavec*wn*T).*sin(sqrt(1-zetavec.^2)*wn*T);
    plot(fx,fy,'Color',[i/length(wnvec),0,0],'LineWidth',2,'DisplayName',"w_n:"+string(wn));
end
legend("show","location","best")
rectangle('Position',[-1,-1,2,2],'Curvature',[1 1]);

print("../../img/"+"lec6_plot1.eps",'-depsc','-r150');
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
figure(2);clf;hold on;grid minor;set(gca, 'MinorGridColor', 'k','MinorGridAlpha',1);
xlabel("Re");ylabel("Im");title("\zeta sabit 0<w_n<100");
T=1;
zetavec=0.1;
wnvec=1:0.1:100;
for i=1:length(zetavec)
    zeta=zetavec(i);
    fx=exp(-zeta*wnvec*T).*cos(sqrt(1-zeta.^2)*wnvec*T);
    fy=exp(-zeta*wnvec*T).*sin(sqrt(1-zeta.^2)*wnvec*T);
    plot(fx,fy,'Color',[i/length(zetavec),0,0],'LineWidth',2,'DisplayName',"\zeta:"+string(zeta));
end
legend("show","location","best")
rectangle('Position',[-1,-1,2,2],'Curvature',[1 1]);
print("../../img/"+"lec6_plot2.eps",'-depsc','-r150');
figure(3);clf;hold on;grid minor;set(gca, 'MinorGridColor', 'k','MinorGridAlpha',1);
xlabel("Re");ylabel("Im");title("\zeta sabit 0<w_n<100");
T=1;
zetavec=0.5;
wnvec=1:0.1:100;
for i=1:length(zetavec)
    zeta=zetavec(i);
    fx=exp(-zeta*wnvec*T).*cos(sqrt(1-zeta.^2)*wnvec*T);
    fy=exp(-zeta*wnvec*T).*sin(sqrt(1-zeta.^2)*wnvec*T);
    plot(fx,fy,'Color',[i/length(zetavec),0,0],'LineWidth',2,'DisplayName',"\zeta:"+string(zeta));
end
legend("show","location","best")
rectangle('Position',[-1,-1,2,2],'Curvature',[1 1]);

print("../../img/"+"lec6_plot3.eps",'-depsc','-r150');
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
figure(4);clf;hold on;grid minor;set(gca, 'MinorGridColor', 'k','MinorGridAlpha',1);
xlabel("Re");ylabel("Im");title("ts sabit");
T=1;
zetavec=0.01:0.01:1;
tsvec=[5,10,30];
for i=1:length(tsvec)
    ts=tsvec(i);
    wn=4./(ts.*zetavec);
    fx=exp(-zetavec.*wn*T).*cos(sqrt(1-zetavec.^2).*wn*T);
    fy=exp(-zetavec.*wn*T).*sin(sqrt(1-zetavec.^2).*wn*T);
    plot(fx,fy,'x','Color',[i/length(tsvec),0,0],'LineWidth',2,'DisplayName',"t_s:"+string(ts));
end
legend("show","location","best")
rectangle('Position',[-1,-1,2,2],'Curvature',[1 1]);
print("../../img/"+"lec6_plot4.eps",'-depsc','-r150');
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

File: lecture notes/python/lec6/run1.py
import matplotlib.pyplot as plt
import numpy as np
import sympy as sym
import tbcontrol
import control
############################
# q1
############################
def fun_q1():
    T=1
    zetavec=np.arange(0.1,1,0.1)
    wnvec=np.arange(1,6,1)

    for i in range(0,len(wnvec)):
        wnval=wnvec[i]
        
        fx=np.zeros(zetavec.shape)
        fy=np.zeros(zetavec.shape)
        for j in range(0,len(zetavec)):
            zetaval=zetavec[j]
            fx[j]=np.exp(-zetaval*wnval*T)*np.cos(np.sqrt(1-zetaval**2)*wnval*T)
            fy[j]=np.exp(-zetaval*wnval*T)*np.sin(np.sqrt(1-zetaval**2)*wnval*T)
    
        plt.plot(fx,fy,label="wn:"+str(wnval),color=((i+1)/len(wnvec),0,0))

    plt.legend(loc="upper left")
    plt.gca().add_patch(plt.Circle((0, 0), 1, color='k',fill=False))
    plt.show()
############################
# q2a
############################
def fun_q2a():
    T=1
    zetaval=0.1
    wnvec=np.arange(1,100+1,0.1)

    fx=np.zeros(wnvec.shape)
    fy=np.zeros(wnvec.shape)
    for i in range(0,len(wnvec)):
        wnval=wnvec[i]
        fx[i]=np.exp(-zetaval*wnval*T)*np.cos(np.sqrt(1-zetaval**2)*wnval*T)
        fy[i]=np.exp(-zetaval*wnval*T)*np.sin(np.sqrt(1-zetaval**2)*wnval*T)

    plt.plot(fx,fy,label="zeta:"+str(zetaval))
    plt.legend(loc="upper left")
    plt.gca().add_patch(plt.Circle((0, 0), 1, color='k',fill=False))
    plt.show()
############################
# q2b
############################
def fun_q2b():
    T=1
    zetaval=0.5
    wnvec=np.arange(1,100+1,0.1)

    fx=np.zeros(wnvec.shape)
    fy=np.zeros(wnvec.shape)
    for i in range(0,len(wnvec)):
        wnval=wnvec[i]
        fx[i]=np.exp(-zetaval*wnval*T)*np.cos(np.sqrt(1-zetaval**2)*wnval*T)
        fy[i]=np.exp(-zetaval*wnval*T)*np.sin(np.sqrt(1-zetaval**2)*wnval*T)

    plt.plot(fx,fy,label="zeta:"+str(zetaval))
    plt.legend(loc="upper left")
    plt.gca().add_patch(plt.Circle((0, 0), 1, color='k',fill=False))
    plt.show()

############################
# q3
############################
def fun_q3():
    T=1
    zetavec=np.arange(0.001,1,0.001)
    tsvec=np.array([5,10,30])

    for i in range(0,len(tsvec)):
        tsval=tsvec[i]
        fx=np.zeros(zetavec.shape)
        fy=np.zeros(zetavec.shape)
        for j in range(0,len(zetavec)):
            zetaval=zetavec[j]
            wnval=4/(tsval*zetaval)
            fx[j]=np.exp(-zetaval*wnval*T)*np.cos(np.sqrt(1-zetaval**2)*wnval*T)
            fy[j]=np.exp(-zetaval*wnval*T)*np.sin(np.sqrt(1-zetaval**2)*wnval*T)
        plt.plot(fx,fy,label="ts:"+str(tsval),marker='x',color=((i+1)/len(tsvec),0,0),linestyle='')
    plt.legend(loc="upper left")
    plt.gca().add_patch(plt.Circle((0, 0), 1, color='k',fill=False))
    plt.show()
fun_q3()